scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -3573,7 +3573,7 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
   }
 
   def ctorModifiers(): Option[Mod] = token match {
-    case Unquote() => Some(unquote[Mod])
+    case Unquote() if ahead(token.is[LeftParen]) => Some(unquote[Mod])
```

Comment:
i am not sure i understand this. does this mean unquoting a mod requires `$(mod)`, with parens?

Reply:
It means as far as I can tell `$mod (` so essentially the constructor should be next.

Reply:
ah, so this is not for `private class C(...)`, it's for `class C private(...)`? got it.